### PR TITLE
`Width` and `Coverage` metrics for prediction intervals

### PR DESCRIPTION
**Pull request by martins0n**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: etna/metrics/intervals_metrics.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        self._validate_segment_columns(y_true=y_true, y_pred=y_pred)
+
+        segments = set(y_true.df.columns.get_level_values("segment"))
+        metrics_per_segment = {}
+        for segment in segments:
+            self._validate_timestamp_columns(
+                timestamp_true=y_true[:, segment, "target"].dropna().index,
+                timestamp_pred=y_pred[:, segment, "target"].dropna().index,
+            )
+            upper_quantile_flag = y_true[:, segment, "target"] <= y_pred[:, segment, f"target_{self.quantiles[1]:.4g}"]
```
Comment (julia-shenshina) [inline etna/metrics/intervals_metrics.py:71]: looks like we need to check that `f"target_{self.quantiles[1]:.4g}"` is in y_pred

Comment (julia-shenshina) [reply etna/metrics/intervals_metrics.py:71]: why should we format quantiles values like `f"target_{self.quantiles[1]:.4g}"`?

Comment (martins0n) [reply etna/metrics/intervals_metrics.py:81]: > looks like we need to check that `f"target_{self.quantiles[1]:.4g}"` is in y_pred

I've added validation for it and test case

Comment (martins0n) [reply etna/metrics/intervals_metrics.py:81]: > why should we format quantiles values like `f"target_{self.quantiles[1]:.4g}"`?

I don't remember exactly where issues could arise. But in some cases due to float point arithmetic we had bad quantile representation, so we've started using such formatting

**Review by julia-shenshina — DISMISSED** [review]

**Review by julia-shenshina — APPROVED** [review]

**Pull request merged**